### PR TITLE
feat(cb2-6303): update dangerous defect taxonomy

### PR DESCRIPTION
**Pull request by Ruairidh-BJSS**
## Update dangerous defect taxonomy

_Modify the defect taxonomy to include an Asterix next to each dangerous defect which does not require the VSA to issue a prohibition_
[link to ticket number](https://dvsa.atlassian.net/browse/CB2-6303)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by tahayazici — APPROVED** [review]

File: src/app/forms/components/defect-select/defect-select.component.html [hunk 1]
```diff
@@ -37,7 +37,9 @@ <h1 class="govuk-heading-l">Select a defect</h1>
           {{ deficiency.deficiencySubId ? '(' + deficiency.deficiencySubId + ')' : '' }}</a
         >
         <span>
-          <app-tag class="browse__category" [type]="categoryColor(deficiency.deficiencyCategory)">{{ deficiency.deficiencyCategory }}</app-tag>
+          <app-tag class="browse__category" [type]="categoryColor(deficiency.deficiencyCategory)"
+            >{{ deficiency.deficiencyCategory }}{{ deficiency.stdForProhibition ? '*' : '' }}</app-tag
```
Comment (gjulien-bjss) [inline src/app/forms/components/defect-select/defect-select.component.html:41]: Don't know how much of a difference it makes, looks a bit cleaner this way
```suggestion
            >{{ deficiency.deficiencyCategory + deficiency.stdForProhibition ? '*' : '' }}</app-tag
```

File: src/app/forms/custom-sections/defect/defect.component.html [hunk 2]
```diff
@@ -1,66 +1,68 @@
...
+  <ng-container *ngIf="defect">
+    <h1 class="govuk-heading-l">Defect details</h1>
+    <div class="govuk-grid-row govuk-!-margin-bottom-5">
+      <div class="govuk-grid-column-two-thirds">
+        <span class="govuk-caption-m">{{ defect.imDescription }}</span>
+        <h2 class="govuk-heading-s">
+          {{ defect.imNumber }}.{{ defect.itemNumber }} {{ defect.deficiencyId ? '(' + defect.deficiencyId + ')' : '' }}
+          {{ defect.deficiencySubId ? '(' + defect.deficiencySubId + ')' : '' }} - {{ defect.itemDescription }} {{ defect.deficiencyText }}
+        </h2>
+        <app-tag [type]="categoryColor(defect.deficiencyCategory)">{{ defect.deficiencyCategory }}{{ isDangerousAsterisk ? '*' : '' }}</app-tag>
```
Comment (gjulien-bjss) [inline src/app/forms/custom-sections/defect/defect.component.html:13]:
```suggestion
        <app-tag [type]="categoryColor(defect.deficiencyCategory)">{{ defect.deficiencyCategory + isDangerousAsterisk ? '*' : '' }}</app-tag>
```

File: src/app/forms/custom-sections/defect/defect.component.html [hunk 3]
```diff
@@ -71,17 +73,17 @@ <h2 class="govuk-heading-m">Defect location</h2>
...
+        <span class="govuk-caption-m">{{ defect.imDescription }}</span>
         <h2 class="govuk-heading-s">
-          {{ defect!.imNumber }}.{{ defect!.itemNumber }} {{ defect!.deficiencyId ? '(' + defect!.deficiencyId + ')' : '' }}
-          {{ defect!.deficiencySubId ? '(' + defect!.deficiencySubId + ')' : '' }} - {{ defect!.itemDescription }} {{ defect!.deficiencyText }}
+          {{ defect.imNumber }}.{{ defect.itemNumber }} {{ defect.deficiencyId ? '(' + defect.deficiencyId + ')' : '' }}
+          {{ defect.deficiencySubId ? '(' + defect.deficiencySubId + ')' : '' }} - {{ defect.itemDescription }} {{ defect.deficiencyText }}
         </h2>
-        <app-tag [type]="categoryColor(defect!.deficiencyCategory)">{{ defect!.deficiencyCategory }}</app-tag>
-        <app-tag *ngIf="defect!.prs" id="prs-display" [type]="'blue'">PRS</app-tag>
+        <app-tag [type]="categoryColor(defect.deficiencyCategory)">{{ defect.deficiencyCategory }}{{ isDangerousAsterisk ? '*' : '' }}</app-tag>
```
Comment (gjulien-bjss) [inline src/app/forms/custom-sections/defect/defect.component.html:85]:
```suggestion
        <app-tag [type]="categoryColor(defect.deficiencyCategory)">{{ defect.deficiencyCategory + isDangerousAsterisk ? '*' : '' }}</app-tag>
```

File: src/app/forms/custom-sections/defects/defects.component.html [hunk 4]
```diff
@@ -29,7 +29,9 @@
               {{ defect.itemDescription + ' ' + defect.imDescription + ' ' + defect.deficiencyText | truncate: [47] }}
             </td>
             <td class="govuk-table__cell">
-              <app-tag [type]="categoryColor(defect.deficiencyCategory)">{{ defect.deficiencyCategory }}</app-tag>
+              <app-tag [type]="categoryColor(defect.deficiencyCategory)"
+                >{{ defect.deficiencyCategory }}{{ defect.stdForProhibition ? '*' : '' }}</app-tag
```
Comment (gjulien-bjss) [inline src/app/forms/custom-sections/defects/defects.component.html:33]:
```suggestion
                >{{ defect.deficiencyCategory +  defect.stdForProhibition ? '*' : '' }}</app-tag
```

**Pull request merged**